libkbfs/disk_block_cache.go

```diff
@@ -271,12 +276,27 @@ func (cache *DiskBlockCacheStandard) syncBlockCountsFromDb() error {
 	return nil
 }
 
-// compactCachesLocked manually forces both the block cache and LRU cache to
-// compact their underlying data.
+// compactCachesLocked manually forces the disk cache databases to compact
+// their underlying data.
 func (cache *DiskBlockCacheStandard) compactCachesLocked(ctx context.Context) {
-	cache.blockDb.CompactRange(util.Range{})
-	cache.metaDb.CompactRange(util.Range{})
-	cache.tlfDb.CompactRange(util.Range{})
+	// Execute these in a goroutine so we don't block reads or the completion
+	// of the delete that called this. Use a sentinel channel to make sure only
+	// one compaction can happen at once.
+	select {
+	case <-cache.compactCh:
+		blockDb := cache.blockDb
+		metaDb := cache.metaDb
+		tlfDb := cache.tlfDb
+		go func() {
+			metaDb.CompactRange(util.Range{})
```

Comment:
Can you add logging here before and after the compactions (maybe before and after each one?) so we can tell how long the compactions take, and when they start/stop, so that we can line that up with CPU graphs?